### ISSUE TITLE
Comment printing with babel AST fails

### ISSUE THREAD

**Issue by cpojer** [benjamn/recast#203]
There are a few issues with printing comments when working with a babel AST.

Example:
```js
/*
 * Multiline
 */
var MyComponent = React.createClass({
  // This is a comment
  render: function() {
    return <div />;
  }
});
```

printing both comments in this example will fail with errors such as this:
```
AssertionError: {type: CommentBlock, value:
     * Multiline
     , start: 113, end: 132, loc: [object Object], range: 113,132, leading: true, trailing: false} does not match type Comment
```

**Comment by benjamn** [comment]
It looks like Babel has adopted better type names for `Comment` nodes, whereas Esprima used `Block` and `Line` (ugh), which is what `ast-types` currently defines: https://github.com/benjamn/ast-types/blob/b263e474802408fccf72b5f331f50498492f21b8/def/core.js#L363-L371

I think the easiest thing to do is to rename `Block` and `Line` to `CommentBlock` and `CommentLine`, and then define the original types as subtypes, for the sake of backwards compatibility.

**Comment by cpojer** [comment]
https://github.com/benjamn/ast-types/pull/120

**Comment by benjamn** [comment]
This should be fixed by c23f7c0f0a6392826357097c90257019ce1498e4 and f7b7be7c2a96c9c785d51798cde440161ad30602.
